src/Constraint/MultiConstraint.php

```diff
@@ -190,25 +193,48 @@ private static function optimizeConstraints(array $constraints, $conjunctive)
     {
         // parse the two OR groups and if they are contiguous we collapse
         // them into one constraint
-        if (!$conjunctive
-            && 2 === count($constraints)
-            && $constraints[0] instanceof MultiConstraint
-            && $constraints[1] instanceof MultiConstraint
-            && 2 === count($constraints[0]->getConstraints())
-            && 2 === count($constraints[1]->getConstraints())
-            && ($a = (string) $constraints[0])
-            && strpos($a, '[>=') === 0 && (false !== ($posA = strpos($a, '<', 4)))
-            && ($b = (string) $constraints[1])
-            && strpos($b, '[>=') === 0 && (false !== ($posB = strpos($b, '<', 4)))
-            && substr($a, $posA + 2, -1) === substr($b, 4, $posB - 5)
-        ) {
-            return array(
-                array(
-                    new Constraint('>=', substr($a, 4, $posB - 5)),
-                    new Constraint('<', substr($b, $posB + 2, -1)),
-                ),
-                true,
-            );
+        // [>= 1 < 2] || [>= 2 < 3] || [>= 3 < 4] => [>= 1 < 4]
+        if (!$conjunctive) {
+            $lc = $constraints[0];
+            $mergedConstraints = array();
+            $optimized = false;
+            for ($i = 1, $l = count($constraints); $i <$l; $i++) {
+                $rc = $constraints[$i];
+                if (
+                    $lc instanceof MultiConstraint
+                    && $lc->conjunctive
+                    && $rc instanceof MultiConstraint
+                    && $rc->conjunctive
+                    && ($lc0 = (string) $lc->constraints[0])
+                    && substr($lc0, 0, 2) === '>='
```

Comment:
Not sure this path is hot enough, but this could be optimized to

```suggestion
                    && $lc[0] === '>' && $lc[1] === '='
```

Reply:
thanks